src/java.xml/share/classes/org/xml/sax/helpers/XMLReaderFactory.java

```diff
@@ -236,21 +234,18 @@ private static String jarLookup(final ClassLoader loader) {
      *
      * @return instance of provider class if found or null
      */
-    @SuppressWarnings("removal")
     private static <T> T findServiceProvider(final Class<T> type, final ClassLoader loader)
             throws SAXException {
         ClassLoader cl = Objects.requireNonNull(loader);
         try {
-            return AccessController.doPrivileged((PrivilegedAction<T>) () -> {
-                final ServiceLoader<T> serviceLoader;
-                serviceLoader = ServiceLoader.load(type, cl);
-                final Iterator<T> iterator = serviceLoader.iterator();
-                if (iterator.hasNext()) {
-                    return iterator.next();
-                } else {
-                    return null;
-                }
-            });
+            final ServiceLoader<T> serviceLoader;
+            serviceLoader = ServiceLoader.load(type, cl);
+            final Iterator<T> iterator = serviceLoader.iterator();
+            if (iterator.hasNext()) {
+                return iterator.next();
+            } else {
+                return null;
+            }
```

Comment:
In passing, the Iterator isn't needed here, this block can be reduced to`return ServiceLoader.load(type, cl).findFirst().orElse(null);`

Reply:
Will update